mlrun/pipelines/__init__.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+from importlib.abc import MetaPathFinder
+from importlib.util import spec_from_file_location
+
+import kfp
+
+
+def resolve_pipeline_engine():
+    if kfp.__version__.startswith("2.0"):
```

Comment:
Maybe `2.0` is too strict? Just thinking that if at some point there'll be a `2.1` version to kfp, it'll probably still need the `kfp-v2.0` module (hopefully it's BC) but this `if` will not catch it and will return `kfp-v1.8`.
Why not being stricter about this check? If it's 2.0, return v2.0, if it's exactly 1.8 return v1.8, else explode. This way at least the developers will be immediately aware when such changes happen.